### PR TITLE
original experiment code

### PR DESCRIPTION
**Pull request by davidterroso**
There are a few concerns with the code:

- It may be computationally expensive to run as the number of batches is bigger, the number of steps is bigger, the number of neurons is also bigger.
- The learning rate is not mentioned in the paper so it was just kept the same
- The time horizon, or number of steps per episode, was fixed at 9 which was the value that obtained better results, as expected
- The accuracy expected is 88%

### CODE REVIEW

**Comment by davidterroso** [comment]
- The function responsible for reading the messages from other agents is implemented

File: marl_classification/networks/ft_extractor.py [hunk 1]
```diff
@@ -47,16 +47,15 @@ def __init__(self, f: int) -> None:
...
             nn.Conv2d(16, 32, (3, 3), padding=1),
             nn.GELU(),
-            nn.MaxPool2d(2, 2),
             nn.BatchNorm2d(32),
+            nn.MaxPool2d(2, 2) if f > 2 else nn.Identity(),
             nn.Flatten(1, -1),
         )
 
-        self.__out_size = 32 * (f // 4) ** 2
+        self.__out_size = 32 * ((f // 2) ** 2 if f > 2 else f ** 2)
```
Comment (seifhussam) [inline marl_classification/networks/ft_extractor.py:58]: any reason why we are introducing changes in the feature extractor?

Comment (davidterroso) [reply marl_classification/networks/ft_extractor.py:58]: The code crashes if it receives a f value lower than 4 because the self.__out_size variable is set to 0. I changed the code so that the the self.__out_size is not set to 0 whenever the f value is lower than 4.
